src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,196 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY;
+    private static final int SOFT_MAX_ARRAY_LENGTH;
+    private static final Object[] DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA;
+    private static final int NOT_EXISTING_INDEX;
 
+    static {
+        NOT_EXISTING_INDEX = -1;
+        DEFAULT_CAPACITY = 10;
+        SOFT_MAX_ARRAY_LENGTH = Integer.MAX_VALUE - 8;
+        DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA = new Object[]{};
     }
 
-    @Override
-    public void add(T value, int index) {
+    private Object[] elementData;
+    private int size;
 
+    public ArrayList() {
+        elementData = DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA;
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
your solution is overcomplicated actually. Let's try to simplify everything. For example, here you can just iterate throughout list and reuse your add() method to add new element to your array